src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -1377,38 +1377,36 @@ Namespace Microsoft.CodeAnalysis.Operations
             Dim constantValue As [Optional](Of Object) = New [Optional](Of Object)()
             Dim isImplicit As Boolean = boundRaiseEventStatement.WasCompilerGenerated
 
+            Dim eventSymbol = boundRaiseEventStatement.EventSymbol
             Dim eventInvocation = TryCast(boundRaiseEventStatement.EventInvocation, BoundCall)
 
             ' Return an invalid statement for invalid raise event statement
-            If eventInvocation Is Nothing OrElse eventInvocation.ReceiverOpt Is Nothing Then
+            If eventInvocation Is Nothing OrElse (eventInvocation.ReceiverOpt Is Nothing AndAlso Not eventSymbol.IsShared) Then
                 Debug.Assert(boundRaiseEventStatement.HasErrors)
                 Dim children As Lazy(Of ImmutableArray(Of IOperation)) = New Lazy(Of ImmutableArray(Of IOperation))(Function() ImmutableArray.Create(Create(boundRaiseEventStatement.EventInvocation)))
                 Return New LazyInvalidOperation(children, _semanticModel, syntax, type, constantValue, isImplicit)
             End If
 
-            Dim receiver = eventInvocation.ReceiverOpt
-            Dim eventSymbol = boundRaiseEventStatement.EventSymbol
-            Dim eventReferenceSyntax = receiver.Syntax
+            Dim receiverOpt = eventInvocation.ReceiverOpt
+            Dim eventReferenceSyntax = If(receiverOpt?.Syntax,
+                                          If(TryCast(syntax, RaiseEventStatementSyntax)?.Name,
+                                             syntax))
```

Comment:
Is there a test hits this line of code? i.e. syntax is not a RaiseEventStatementSyntax.

Reply:
> Is there a test hits this line of code? i.e. syntax is not a RaiseEventStatementSyntax.

I don't think it is possible to get into this situation at the moment, but would like the code to be robust enough.